dali/util/user_stream.h

```diff
@@ -42,12 +42,14 @@ class DLL_PUBLIC UserStream {
 
   DLL_PUBLIC cudaStream_t GetStream(const dali::Buffer<GPUBackend> &b) {
     size_t dev = GetDeviceForBuffer(b);
-    std::unique_lock<std::mutex> lock(m_);
+    std::lock_guard<std::mutex> lock(m_);
     auto it = streams_.find(dev);
     if (it != streams_.end()) {
       return it->second;
     } else {
-      CUDA_CALL(cudaStreamCreateWithFlags(&streams_[dev], cudaStreamNonBlocking));
+      constexpr int kDefaultStreamPriority = 0;
```

Comment:
Shouldnt the default priority be an argument of this function?

Reply:
we could, but after discussing with Janusz we decided to let the default priority for those user streams (used when invoking `as_cpu` for instance)

Reply:
Sure